device_backends/uio/src/UioAccess.cc

```diff
@@ -32,14 +34,13 @@ namespace ChimeraTK {
   }
 
   void UioAccess::open() {
-    UioMMap();
+    _mmio = std::make_unique<MmioAccess>(_deviceFileDescriptor, _deviceMemSize, true);
```

Comment:
Why is the MmioAccesso class designed such that you have to use a unique_ptr? Wouldn't it make more sense to use a plain instance and call some init function here?

Reply:
TBH I don't remember

Reply:
Still, I find it a bit strange to make an object residing in dynamic memory so you can call the constructor and destructor at defined points in time, and then pack the dynamic object into some container to make sure it does not leak.

If you had separate member functions to control the required actions at open and close and just call close in the destructor you could get away without dynamic memory allocation and containers. RAII is a nice concept in other context, but here you merely want to split off certain code into its own class. I don't see the benefit of having a different lifetime of the new object than its parent.